File: app/src/main/java/com/cn29/aac/ui/common/ActivityPermissionComponentBuilder.java
package com.cn29.aac.ui.common;

import android.support.v7.app.AppCompatActivity;

public class ActivityPermissionComponentBuilder {

  private final AppCompatActivity activity;
  private String[] permissions;
  private int requestCode;

  public ActivityPermissionComponentBuilder(AppCompatActivity activity) {
    this.activity = activity;
  }


  public ActivityPermissionComponentBuilder setPermissions(String[] permissions) {
    this.permissions = permissions;
    return this;
  }

  public ActivityPermissionComponentBuilder setRequestCode(int requestCode) {
    this.requestCode = requestCode;
    return this;
  }

  public ActivityPermissionComponent createActivityPermissionComponent() {
    return new ActivityPermissionComponent(activity, permissions, requestCode);
  }
}

File: app/src/main/java/com/cn29/aac/ui/common/ActivityPermissionComponentBuilder.kt
package com.cn29.aac.ui.common

import androidx.appcompat.app.AppCompatActivity

class ActivityPermissionComponentBuilder(private val activity: AppCompatActivity) {
    private lateinit var permissions: Array<String>
    private var requestCode = 0
    fun setPermissions(permissions: Array<String>): ActivityPermissionComponentBuilder {
        this.permissions = permissions
        return this
    }

    fun setRequestCode(requestCode: Int): ActivityPermissionComponentBuilder {
        this.requestCode = requestCode
        return this
    }

    fun createActivityPermissionComponent(): ActivityPermissionComponent {
        return ActivityPermissionComponent(activity, permissions, requestCode)
    }

}